src/os_win32.c

```diff
@@ -5186,10 +5186,22 @@ mch_system_piped(char *cmd, int options)
 			// now put 9 as SIGKILL
 			    TerminateProcess(pi.hProcess, 9);
 			}
-			if (ta_buf[ta_len] == Ctrl_D)
+		    }
+
+		    /*
+		     * Check for CTRL-D: EOF, close pipe to child.
+		     * Ctrl_D may be decorated by _OnChar()
+		     */
+		    if ((len == 1 || len == 4 ) && cmd != NULL)
+		    {
+			if (ta_buf[0] == Ctrl_C
```

Comment:
Is this correct?  Not Ctrl_D?
Additionally, the original code used `ta_len` for the index.
```suggestion
			if (ta_buf[ta_len] == Ctrl_D
```

Reply:
My mistake! It must be `Ctrl_D`.